guidance_plugin_validator/src/guidance_plugin_validator/guidance_plugin_components.py

```diff
@@ -0,0 +1,402 @@
+            rospy.loginfo("Success: " + str(self.plugin_name) + " Has plugin discovery message 'activated' field set to True")
+        else:
+            rospy.logwarn("WARNING: " + str(self.plugin_name) + " Does not have plugin discovery message 'activated' field set to True.")
+
+class RequiredStrategicComponents():
+    """
+    Convenience class used to store the validation results for components of a strategic plugin
+    designated as required.
+    """
+
+    def __init__(self, plugin_name):
+        """Constructor for RequiredStrategicComponents"""
+
+        # Validation result indicating whether strategic plugin's node successfully launches
+        self.has_node = False
+
+        # Validation result indicating whether strategic plugin's node advertises required service
+        self.plan_maneuvers_service = "/guidance/plugins/" + str(plugin_name) + "/plan_maneuvers"
+        self.has_plan_maneuvers_service = False
+
+        # Validation result indicating whether strategic plugin's node publishes to required topic
+        self.plugin_discovery_topic = "/guidance/plugin_discovery"
+        self.has_plugin_discovery_pub = False
+
+        # Validation results indicating whether strategic plugin's node publishes required information to the plugin_discovery topic
+        self.correct_plugin_discovery_type = 1
+        self.has_correct_plugin_discovery_type = False
+
+        self.correct_plugin_discovery_capability = "strategic_plan/plan_maneuvers"
+        self.has_correct_plugin_discovery_capability = False
+
+class OptionalStrategicComponents():
+    """
+    Convenience class used to store the validation results for components of a strategic plugin
+    designated as optional.
+    """
+
+    def __init__(self):
+        """Default constructor for RequiredStrategicComponents"""
+
+        # Validation results indicating whether strategic plugin's node subscribes to optional (but commonly useful) topic
+        self.current_pose_topic = "/localization/current_pose"
+        self.has_current_pose_sub = False
+
+        self.current_speed_topic = "/hardware_interface/vehicle/twist"
+        self.has_current_speed_sub = False
+
+        # Validation results indicating whether strategic plugin's node publishes optional (but commonly useful) information to the plugin_discovery topic
+        self.has_plugin_discovery_available = False
+        self.has_plugin_discovery_activated = False
+
+class TacticalPluginResults:
+    """
+    Convenience class used to store the validation results (both required and optional) 
+    for a Guidance Tactical Plugin. 
+    """
+
+    def __init__(self, plugin_name):
+        """Constructor for TacticalPluginResults"""
+
+        # Set the plugin's name, as used by the plugin_discovery topics and the plugin's advertised service(s)
+        self.plugin_name = plugin_name
+        
+        # Set the plugin's node name
+        if plugin_name == "InLaneCruisingPlugin":
```

Comment:
Like above, I'd prefer to use the intended topic patterns instead of specific casing on the plugin names where possible.